source/fundamentals/connection/socks.txt

```diff
@@ -0,0 +1,126 @@
+.. _java-connect-socks:
+
+==========================================
+Connect to MongoDB by Using a SOCKS5 Proxy
+==========================================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to connect to MongoDB by using a SOCKS5 proxy.
+SOCKS5 is a standardized protocol for communicating with network services
+through a proxy server.
+
+.. tip::
+
+   To learn more about the SOCKS5 protocol, see the Wikipedia entry on
+   :wikipedia:`SOCKS <w/index.php?title=SOCKS&oldid=1160087146>`.
+
+.. _socks-proxy-settings:
+
+SOCKS5 Proxy Settings
+---------------------
+
+The proxy settings specify the SOCKS5 proxy server address and your
+authentication credentials. You can specify your settings in an instance of
+``MongoClientSettings`` or in your connection URI.
+
+The following table describes the SOCKS5 client options:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 15 20 65
+
+   * - Name
+     - Accepted Values
+     - Description
+
+   * - **proxyHost**
+     - String
+     - *Required*. Specifies the SOCKS5 proxy IPv4 address, IPv6 address, or hostname.
+
+   * - **proxyPort**
+     - non-negative Integer
+     - Specifies the TCP port number of the SOCKS5 proxy server. If you
+       set the ``proxyHost`` option, the value of this option defaults
+       to ``1080``.
+
+   * - **proxyUsername**
+     - String
+     - Specifies the username for authentication to the SOCKS5
+       proxy server. The driver ignores ``null`` and empty string values for
+       this setting.
+
+   * - **proxyPassword**
+     - String
+     - Specifies the password for authentication to the SOCKS5
+       proxy server. The driver ignores ``null`` and empty string values
+       for this setting.
+
+
+Examples
+--------
+
+The following examples show how to instantiate a ``MongoClient`` that connects
+to MongoDB by using a SOCKS5 proxy by using a ``MongoClientSettings`` instance
+or a connection string. These examples use the placeholder values described in
```

Comment:
Suggestion: The repeated phrase: "by using ... by using" made this sentence hard to understand on my first read. I think we could either split the first sentence into two, so something like:

```suggestion
The following examples show how to instantiate a ``MongoClient`` that connects
to MongoDB by using a SOCKS5 proxy.  The proxy settings can be specified in the ``MongoClientSettings`` instance
or a connection string. These examples use the placeholder values described in
```

Or instead of splitting, maybe adding more detail will make it more clear:

```suggestion
The following examples show how to instantiate a ``MongoClient`` that connects
to MongoDB by using a SOCKS5 proxy and specifying the proxy settings in the ``MongoClientSettings`` instance
or in a connection string. These examples use the placeholder values described in
```

Reply:
Great suggestion, thanks!